classes/02class/exercises/c02-network12-challenge/alanlima/README.md

```diff
@@ -0,0 +1,138 @@
+# c02-network12
+
+## Commands Execution Output
+
+For the exercises below, you should use all the custom created network resources. NOT THE DEFAULT ONES.
+
+- Provide commands used to go through all the below deliverables: (everything should be done using the AWS cli)
+
+  - **explain how was the process to create a VPC peering**
+    1. Create a VPC Peering Connection using the requester VPC ID and the accepter Account ID / VPC ID
+        ```bash
+        $ aws ec2 create-vpc-peering-connection \
+            --peer-owner-id 438549961569 \
+            --peer-vpc-id vpc-081afcee350babdf5 \
+            --vpc-id vpc-08fb60055e82efe81
+            
+        VpcPeeringConnection:
+          AccepterVpcInfo:
+            OwnerId: '438549961569'
+            Region: ap-southeast-2
+            VpcId: vpc-081afcee350babdf5
+          ExpirationTime: '2020-07-11T10:02:28+00:00'
+          RequesterVpcInfo:
+            CidrBlock: 172.32.0.0/16
+            CidrBlockSet:
+            - CidrBlock: 172.32.0.0/16
+            OwnerId: '097922957316'
+            PeeringOptions:
+              AllowDnsResolutionFromRemoteVpc: false
+              AllowEgressFromLocalClassicLinkToRemoteVpc: false
+              AllowEgressFromLocalVpcToRemoteClassicLink: false
+            Region: ap-southeast-2
+            VpcId: vpc-08fb60055e82efe81
+          Status:
+            Code: initiating-request
+            Message: Initiating Request to 438549961569
+          Tags: []
+          VpcPeeringConnectionId: pcx-0107f45bf4b7e1ea9
+        ```
+    2. The accepter should accept the VPC Peering Connection
+        ```bash
+        $ aws ec2 accept-vpc-peering-connection \
+            --vpc-peering-connection-id pcx-0107f45bf4b7e1ea9
+        ```
+    3. Share the public key pair to be used on the requester EC2 instance which will be connected via SSH.
+    4. Update the route table used by the requester VPC ID to add the IP Range from the accepter VPC ID and the Peering Connection ID.
+    
+        4.1 The Requester should also update its route table refering to the IP Range from Accepter VPC ID
+        
+    5. Update the security group from requester to accept SSH (22) connection from accepter IP Range or specific EC2 Instance IP Address.
+    
+
+  - **add routes from origin VPC to target VPC - only for private subnets**
+  
+    ```bash
+    # Requester
+    $ aws ec2 create-route \
+        --destination-cidr-block 172.16.0.0/16 \
+        --vpc-peering-connection-id pcx-094d89b026a8b0740 \
+        --route-table-id rtb-0fcbbbb1627418bed
+    
+    # Accepter
+    $ aws ec2 create-route \
+        --destination-cidr-block 172.32.0.0/16 \
+        --vpc-peering-connection-id pcx-094d89b026a8b0740 \
+        --route-table-id rtb-0f5b2e7a42466bb7d
+    ```
+        
+  - **open any network ACL to allow the connection within the subnets -- try to   only open the mininum required**
+  
+     ```bash
+     $ aws ec2 create-network-acl-entry \
```

Comment:
are you sure it worked only opening the port 22 ingress?

Reply:
We missed that, it was working because the next rule was allowing everything 🙄
I just updated the rules to be more restrictive.

Reply:
the fix was actually adding the egress rules that was missing